File: 11722/main.cpp
#include <iostream>
#include <vector>

using namespace std;

int main (void) {

	int n;
	cin >> n;
	vector<int> arr(n);
	vector<int> dp(n);

	for (int i = 0; i < n ; i++) cin >> arr[i];

	for (int i = 0; i < n ; i++ ) {
		dp[i] = 1;
		for (int j = 0; j <	i ; j++ ) {
			if ( arr[i] < arr[j] && dp[j] + 1 > dp[i] ) 
				dp[i] = dp[j] + 1;
		}
	}

	int max = -1;

	for (int i = 0; i < n ; i++ ) {
		if ( dp[i] > max ) 
			max = dp[i];
	}		

	cout << max << endl;

	return 0;
}	

File: 11722/Main.java
import java.util.Scanner;

public class Main {
	public static void main(String [] args) {

		Scanner sc = new Scanner(System.in);

		final int MAX = 1001;
		int n;
		int arr[] = new int[MAX];
		int dp[] = new int[MAX];

		n = sc.nextInt();

		for ( int i = 1 ; i <= n ; i++ )	
			arr[i] = sc.nextInt();

		int ans = -1;

		for ( int i = 1; i <= n ; i++ ) {
			dp[i] = 1;
			for ( int j = 1 ; j <= i ; j++ ) {
				if ( dp[i] < dp[j] + 1 && arr[i] < arr[j] ) 
					dp[i] = dp[j] + 1;
			}
			if ( ans < dp[i] ) 	ans = dp[i];
		}

		System.out.print(ans);
	}
}
